### PR TITLE
feat: highlight the searched text in memo content

### PR DESCRIPTION
**Pull request by Zeng1998**
https://user-images.githubusercontent.com/31177490/203886819-bd540dad-8713-4c8e-8aa3-76bfb551b115.mp4

### CODE REVIEW

**Comment by Mahoo12138** [comment]
But `CRTL+F` also can do this.

File: web/src/components/MemoContent.tsx [hunk 1]
```diff
@@ -5,13 +5,15 @@ import Icon from "./Icon";
 import { SETTING_IS_FOLDING_ENABLED_KEY, IS_FOLDING_ENABLED_DEFAULT_VALUE } from "../helpers/consts";
 import useLocalStorage from "../hooks/useLocalStorage";
 import "../less/memo-content.less";
+import highlightWithWord from "../labs/highlighter";
```
Comment (boojack) [inline web/src/components/MemoContent.tsx:8]: Fix the import order. `.less` should be the last one.

File: web/src/labs/highlighter/index.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    node.parentNode?.insertBefore(span, node);
+    node.parentNode?.removeChild(node);
+  }
+  for (const child of Array.from(node.childNodes)) {
+    dfsAndHighlight(<HTMLElement>child, keyword);
+  }
+  return node.innerHTML;
+}
+
+const highlightWithWord = (html: string, keyword: string | undefined): string => {
```
Comment (boojack) [inline web/src/labs/highlighter/index.ts:18]:
```suggestion
const highlightWithWord = (html: string, keyword?: string): string => {
```

File: web/src/labs/highlighter/index.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+const escapeRegExp = (str: string): string => {
+  return str.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
+};
+
+function dfsAndHighlight(node: HTMLElement, keyword: string) {
```
Comment (boojack) [inline web/src/labs/highlighter/index.ts:5]: The function name is a bit non-standard, how about `walkthroughNodeWithKeyword`?

**Review by boojack — APPROVED** [review]
LGTM

**Pull request merged**